com.avaloq.tools.ddk.test.ui/src/com/avaloq/tools/ddk/test/ui/swtbot/util/ProblemsViewTestUtil.java

```diff
@@ -0,0 +1,166 @@
+/*******************************************************************************
+ * Copyright (c) 2016-2018 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.test.ui.swtbot.util;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swtbot.eclipse.finder.matchers.WidgetMatcherFactory;
+import org.eclipse.swtbot.eclipse.finder.waits.Conditions;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTable;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTree;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+
+import com.avaloq.tools.ddk.test.ui.swtbot.CoreSwtbotTools;
+import com.avaloq.tools.ddk.test.ui.swtbot.SwtWorkbenchBot;
+
+
+/**
+ * Utility class with handy methods for testing Problems view.
+ */
+public final class ProblemsViewTestUtil {
+  public static final String PROBLEMS_VIEW = "org.eclipse.ui.views.ProblemView"; //$NON-NLS-1$

```

Comment:
I would suggest `import static org.eclipse.ui.IPageLayout.ID_PROBLEM_VIEW` instead.

Other button/menu/etc. label strings are probably also similarly available from elsewhere. Probably in some Messages class and hopefully not an internal non-API thing. Grepcode and the eclipse Github repos help quite a bit for finding these.

Reply:
ID_PROBLEM_VIEW: Done.
Others: The relevant strings in org.eclipse.ui.views.markers.internal.MArkerMessages are non-API. So I've stuck with the magic strings.